src/dao/MoodDao.java

```diff
@@ -45,6 +89,34 @@ public MoodDao() {
      * executing the query
      */
     public Mood getMoodById(int moodId) throws DaoException {
-        return null;
+        BufferedReader reader = null;
+        Mood rtnValue = null;
+        try {
+            String ln = "";
+            reader = new BufferedReader(new FileReader("data/Moods.csv"));
+            reader.readLine();
+
+            while ((ln = reader.readLine()) != null) {
+                String[] obj = ln.split(COMMA_DELIMITER);
+                if (obj.length > 0) {
+                    Mood md = new Mood(Integer.parseInt(obj[MOOD_ID_INDEX]), obj[MOOD_NAME_INDEX]);
+                    if(md.getId() == moodId){
+                        rtnValue = md;
+                    }
+                }
+            }
+            return rtnValue;
+        } catch (Exception e) {
```

Comment:
This should catch specific exceptions. NetBeans will fix it for you if you place your cursor in the Exception fiel, press alt-enter, and click the "replace with…" suggestion.